cpu/riscv_common/irq_arch.c

```diff
@@ -83,20 +87,45 @@ void riscv_irq_init(void)
 /**
  * @brief Global trap and interrupt handler
  */
-__attribute((used))
-static void handle_trap(uword_t mcause)
+__attribute((used)) static void handle_trap(uword_t mcause)
 {
     /*  Tell RIOT to set sched_context_switch_request instead of
      *  calling thread_yield(). */
     riscv_in_isr = 1;
 
     uword_t trap = mcause & CPU_CSR_MCAUSE_CAUSE_MSK;
 
+#ifdef DEVELHELP
+    printf("Trap: mcause=0x%" PRIx32 " mepc=0x%lx mtval=0x%lx\n",
+           (uint32_t)mcause, read_csr(mepc), read_csr(mtval));
+
+    if ((mcause & ~MCAUSE_INT) <= 0xb) {
+        const char *error_messages[] = {
+            "Instruction alignment: Does not occur on RP2350, because 16-bit compressed instructions are implemented, and it is impossible to jump to a byte-aligned address.",
```

Comment:
```suggestion
            "Instruction alignment: Does not occur on RP2350, because "
            "16-bit compressed instructions are implemented, and it is "
            "impossible to jump to a byte-aligned address.",
```

Perhaps you can split the other lines up too? C allows you to do that:
https://stackoverflow.com/a/797351

Reply:
Updated the entire function to be a bit less hacky (The error messages were straight out of the rp2350 documentation, the new one follows the official RISC-V specs https://riscv.github.io/riscv-isa-manual/snapshot/privileged/#mcause)